### DISCUSSION TITLE
Insert data without column name

### DISCUSSION THREAD

**Discussion by Philippe-M** [catfan/Medoo#1092] [General]
Hello,

Is it possible to use :

`$database-> insert ("table", [])`

without knowing indicating field names?

Thanks

**Comment by catfan** [comment]
I don't understand this question.

**Reply by Philippe-M** [reply]
Hi,

Currently to use insert() you have to give the table, the name of the columns and their values.

I am looking for a way to insert data without giving the names of the columns.

Example :

```
tbToto {
  f1 varchar(10),
  f2 varchar(10),
  f3 varchar(10)
}
```

with insert() the code would look like

```
$database->insert("tbToto", [
  "f1" => "val1",
  "f2" => "val2",
  "f3" => "val3"
]);
```

I'm trying to do:

$database->insert("tbToto", [ "val1", "val2", "val3" ]);

What would give in SQL

`INSERT INTO tbToto VALUES("val1", "val2", "val3");`

In this way I don't have to write the structure of my table to make the insertion, just given the good number of value and in the good orde.

I use this to make mass insertion of data.

```
      for($i=0; $i<=$nb; $i+=10) {
        $values = array_slice($data, $i, 10, false);
        $database->query('INSERT INTO '.$table.'_tmp VALUES '.implode(', ', $values));
      }
```

Thanks.

**Comment by catfan** [comment]
I don't think this is a good idea to do that...🥲
